plugins/woocommerce/client/blocks/assets/js/blocks/payment-method-icons/index.tsx

```diff
@@ -0,0 +1,22 @@
+/**
+ * External dependencies
+ */
+import { BlockConfiguration, registerBlockType } from '@wordpress/blocks';
+import { Icon } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import metadata from './block.json';
+import { paymentMethodsIcon } from './icon';
+import edit from './edit';
+import './style.scss';
+
+registerBlockType( metadata.name, {
+	...( metadata as BlockConfiguration ),
+	icon: {
+		src: <Icon icon={ paymentMethodsIcon } />,
+	},
+	edit,
+	save: () => null,
+} );
```

Comment:
Minor but `registerBlockType` also allows for:

```suggestion
registerBlockType( metadata, {
	icon: {
		src: <Icon icon={ paymentMethodsIcon } />,
	},
	edit,
	save: () => null,
} );
```

which feels simpler for me but I don't have strong opinion, just sharing the option.

Reply:
Looks good to me!